src/python/pants/util/process_handler.py

```diff
@@ -106,7 +106,7 @@ def join_and_get_output():
 
 
 def _tee(infile, outfile, return_function):
-  accumulator = StringIO.StringIO()
+  accumulator = io.BytesIO()
```

Comment:
`StringIO.StringIO()` allows for both bytes and unicode. The library was removed in Python3.

Instead, we're using `io.BytesIO()`, which allows for only bytes. This is what we want - `io.StringIO()` fails the tests whereas this passes.